server/vulnerabilities/oval/parsed/definition.go

```diff
@@ -45,19 +56,35 @@ func (r Definition) CollectTestIds() []int {
 	return results
 }
 
-func evalCriteria(c *Criteria, testResults map[int][]fleet.Software) bool {
+func evalCriteria(c *Criteria, OSTstResults map[int]bool, pkgTstResults map[int][]fleet.Software) (bool, error) {
 	var vals []bool
 	var result bool
 
 	for _, co := range c.Criteriums {
-		r := len(testResults[co]) > 0
-		vals = append(vals, r)
+		pkgTstR, pkgOk := pkgTstResults[co]
+		if pkgOk {
+			vals = append(vals, len(pkgTstR) > 0)
+		}
+
+		OSTstR, OSTstOk := OSTstResults[co]
+		if OSTstOk {
+			vals = append(vals, OSTstR)
+		}
+
+		if !pkgOk && !OSTstOk {
+			return false, fmt.Errorf("test not found: %d", co)
+		}
 	}
+
 	result = c.Operator.Eval(vals...)
 
 	for _, ci := range c.Criterias {
-		return c.Operator.Eval(result, evalCriteria(ci, testResults))
```

Comment:
This was a silly bug I found while testing RHEL - the bug didn't show itself when testing on Ubuntu because the criteria trees were really simple